src/Form/AdminSettingsForm.php

```diff
@@ -76,17 +76,31 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#empty' => $this->t('There are no items yet. Add roles.'),
       '#tabledrag' => [
         [
+          'table_id' => 'edit-role-theme-switcher',
           'action' => 'order',
           'relationship' => 'sibling',
           'group' => 'role-order-weight',
         ],
       ],
     ];
 
-    // Generates a form element for each role.
-    $roles = $this->getUserRoles();
-    foreach ($roles as $rid => $role) {
+    $this->setRowPerRole($this->getUserRoles(), $form);
+
+    $form['actions']['submit']['#tableselect'] = TRUE;
+    return $form;
+  }
 
+  /**
+   * Generate one row into form table per role.
+   *
+   * @param array $roles
+   *   An array contain all roles datas values.
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   */
+  private function setRowPerRole(array $roles, array &$form) {
```

Comment:
I don't agree to move this part in an other function. The code is not complicated enough to require this, and this does not bring a better code readability.